src/java.base/share/classes/java/lang/StringUTF16.java

```diff
@@ -1630,6 +1626,12 @@ static int getChars(long i, int index, byte[] buf) {
         }
         return charPos;
     }
+
+    private static int inflatePacked(int v) {
+        int packed = (int) StringLatin1.PACKED_DIGITS[v];
+        return ((packed & 0xFF) << HI_BYTE_SHIFT)
```

Comment:
I'm not sure this is correct.

Compare `StringUTF16::putChar` where these constants are used to shift _right_ to extract the equivalent byte from a value:
```
        val[index++] = (byte)(c >> HI_BYTE_SHIFT);
        val[index]   = (byte)(c >> LO_BYTE_SHIFT);
```
I.e., when inflating a `byte` `0xaa` to a `char` we end up with 0xaa00 on big-endian.

Since `HI_BYTE_SHIFT` is 8 on big-endian and 0 on little-endian I guess this might work:

```return ((packed & 0xFF) << 16 + HI_BYTE_SHIFT) | ((packed & 0xFF00) << HI_BYTE_SHIFT)```

.. but we really need to prototype and test this out thoroughly on a big-endian system. I second @RogerRiggs notion that the best course of action right now is to back out #14699 and redo it with big-endianness issues resolved.

Reply:
I'm also not sure if this PR is correct.